pandas/core/generic.py

```diff
@@ -4850,7 +4850,7 @@ def _where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
 
         msg = "Boolean array expected for the condition, not {dtype}"
 
-        if not isinstance(cond, pd.DataFrame):
+        if not isinstance(cond, (pd.DataFrame, pd.Panel)):
```

Comment:
we are not supporting Panels much anymore (its going to be deprecated soon anyhow). I don't want to have to deal with this here.